coconut/control/control.go

```diff
@@ -262,56 +265,62 @@ func GetEnvironments(cxt context.Context, rpc *coconut.RpcClient, cmd *cobra.Com
+		return fmt.Errorf("cannot get configuration payload value: %w", err)
 	}
 	userWfPath, err := cmd.Flags().GetString("workflow-template")
 	if err != nil {
-		err = errors.New("cannot get workflow template value")
-		return
+		return fmt.Errorf("cannot get workflow template value: %w", err)
 	}
 	if cmd.Flags().Changed("configuration") && len(configPayload) == 0 && cmd.Flags().Changed("workflow-template") && len(userWfPath) == 0 {
-		err = errors.New("no configuration payload or workflow template provided")
-		return
+		return fmt.Errorf("no configuration payload or workflow template provided: %w", err)
 	}
 
 	payloadData := new(ConfigurationPayload)
 	var wfPath string
 	if cmd.Flags().Changed("configuration") && len(configPayload) > 0 {
-		configPayloadDoc, err := os.ReadFile(configPayload)
-		if err != nil {
-			err = errors.New("cannot read configuration payload file")
-			return err
-		}
-		err = json.Unmarshal(configPayloadDoc, &payloadData)
-		if err != nil {
-			err = errors.New("cannot unmarshal configuration payload")
-			return err
+		if strings.HasSuffix(strings.ToLower(configPayload), ".json") {
+			configPayloadDoc, err := os.ReadFile(configPayload)
+			if err != nil {
+				return fmt.Errorf("cannot read local configuration payload: %w", err)
+			}
+			err = json.Unmarshal(configPayloadDoc, &payloadData)
+			if err != nil {
+				return fmt.Errorf("cannot unmarshal local configuration payload: %w", err)
+			}
+		} else {
+			if strings.HasPrefix(configPayload, HLCONFIG_PATH_PREFIX+HLCONFIG_COMPONENT_PREFIX) {
+				configPayload = strings.TrimPrefix(configPayload, HLCONFIG_PATH_PREFIX+HLCONFIG_COMPONENT_PREFIX)
+			}
+			configPayloadDoc, err := apricot.Instance().GetRuntimeEntry(HLCONFIG_COMPONENT_PREFIX, configPayload)
+			if err != nil {
+				return fmt.Errorf("cannot retrieve file from "+HLCONFIG_PATH_PREFIX+HLCONFIG_COMPONENT_PREFIX+": %w", err)
+			}
+			err = json.Unmarshal([]byte(configPayloadDoc), &payloadData)
+			if err != nil {
+				return fmt.Errorf("cannot unmarshal remote configuration payload: %w", err)
+			}
 		}
 
 		if cmd.Flags().Changed("workflow-template") && len(userWfPath) == 0 {
 			if len(payloadData.Workflow) > 0 {
 				wfPath = payloadData.Workflow
 			} else {
-				err = errors.New("empty workflow template provided")
-				return err
+				return errors.New("empty workflow template provided")
 			}
 		} else if cmd.Flags().Changed("workflow-template") && len(userWfPath) > 0 {
 			wfPath = userWfPath
 		} else {
 			if len(payloadData.Workflow) > 0 {
 				wfPath = payloadData.Workflow
 			} else {
-				err = errors.New("no workflow template provided in config file")
-				return err
+				return errors.New("no workflow template provided in config file")
 			}
 		}
 	} else {
 		if cmd.Flags().Changed("workflow-template") && len(userWfPath) > 0 {
 			wfPath = userWfPath
 		} else {
-			err = errors.New("empty workflow template provided")
-			return err
+			return errors.New("empty workflow template provided")
```

Comment:
This is really minor, but generally we should avoid having identical log messages (or identical error messages) arising in different spots, because this makes it hard to understand *where* something failed.
Here we are in the "no -c, empty -w" scenario, and the message is fine, but could you please modify the instance of this message on L308 so that it makes it clear that an empty -w was passed as override to a configuration payload, and that this is empty.